specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+event loss during the transition.
+
+### Subscription Lifecycle
+
+The lifecycle of a Subscription is independent of the channel it refers to in
+its `spec.channel` field. The `spec.channel` object reference may refer to
+either an `messaging.knative.dev/v1` Channel resource, or another resource which
+meets the `spec.subscribers` and `spec.delivery` required elements in the
+Channelable duck type. If the referenced `spec.channel` does not currently exist
+or its `Ready` condition is not `true`, then the Subscription's `Ready`
+condition MUST NOT be `true`, and the reason SHOULD indicate that the
+corresponding channel is missing or not ready.
+
+The Subscription MUST also set the `status.physicalSubscription` URIs by
+resolving the `spec.subscriber`, `spec.reply`, and
+`spec.delivery.deadLetterSink` as described in
+[Addressable resolution](#addressable-resolution) before setting the `Ready`
+condition to `true`. If any of the addressable fields fails resolution, the
+Subscription MUST set the `Ready` condition to `false`, and at least one
+condition should indicate the reason for the error. (It is acceptable for none
+of the `spec.subscriber`, `spec.reply`, and `spec.delivery.deadLetterSink`
+fields to contain a `ref` field.)
+
+At least one of `spec.subscriber` and `spec.reply` MUST be set; if only
+`spec.reply` is set, the behavior is equivalent to setting `spec.subscriber`
+except that the Channel SHOULD NOT advertise the ability to process replies
+during the delivery.
+
+Once created, the Subscription's `spec.channel` SHOULD NOT permit updates; to
+change the `spec.channel`, the Subscription can be deleted and re-created. This
+pattern is chosen to make it clear that changing the `spec.channel` is not an
+atomic operation, as it may span multiple storage systems. Changes to
+`spec.subscriber`, `spec.reply`, `spec.delivery` and other fields SHOULD be
+permitted, as these could occur within a single storage system.
+
+When a Subscription becomes associated with a channel (either due to creating
+the Subscription or the channel), the Subscription MUST only set the `Ready`
+condition to `true` after the channel has been configured to send all future
+events to the Subscriptions `spec.subscriber`. The Channel MAY send some events
+to the Subscription before prior to the Subscription's `Ready` condition being
+set to `true`. When a Subscription is deleted, the Channel MAY send some
+additional events to the Subscription's `spec.subscriber` after the deletion.
+
+<!--
+TODO: channel-compatible CRDs (Channelable)
+-->
+
+### Addressable Resolution
+
+Both Trigger and Subscription have optional object references (`ref` in
+`spec.subscriber`, `spec.delivery.deadLetterSink`, and `spec.reply` for
+Subscription) which are expected to conform to the Addressable partial schema
+("duck type"). An object conforms with the Addressable partial schema if it
+contains a `status.address.url` field containing a URL which may be used to
```

Comment:
s/may/MAY/
But why is this a MAY instead of MUST ?
And ... attempt to deliver.... instead of just "deliver"